### ISSUE TITLE
Data type of valueStart

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1032]
**Reported by anonymous on 18 Mar 2013 08:21 UTC**
The type of variable `valueStart` in function `ModelicaInternal_setenv` of ModelicaInternal.c should be changed from `int` to `size_t`.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1032

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 18 Mar 2013 18:23 UTC**
Three more compiler issues on the same context, but this time in ModelicaStrings.c.
```
1>Compiling...
1>ModelicaStrings.c
1>e:\modelicastrings.c(47) : warning C4267: 'initializing' : conversion from 'size_t' to 'int', possible loss of data
1>e:\modelicastrings.c(80) : warning C4267: 'return' : conversion from 'size_t' to 'int', possible loss of data
1>e:\modelicastrings.c(178) : warning C4244: 'return' : conversion from '__w64 int' to 'int', possible loss of data
```

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 26 Mar 2013 13:04 UTC**
Fixed in 653906df8e69e71fd819a3243fcd4d6ad2a8e04d by changing the type int to size_t at two places and by adding two (int) casts to avoid warning messages of the C-compiler.

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 26 Mar 2013 16:36 UTC**
There is now a conflict between signed and unsigned in `ModelicaStrings_substring` when comparing `endIndex > len1` and in the format specifier `%d`. Better stick to `int` for `len1` and use the attached patch.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 26 Mar 2013 18:54 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 27 Mar 2013 07:22 UTC**
Fixed in d42440cfddbeabc4df6b8fe0d2ba4c3ccace3bc5 according to the suggestion in the patch

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 27 Mar 2013 07:22 UTC**
